pumpkin-protocol/src/server/play/s_update_sign.rs

```diff
@@ -0,0 +1,37 @@
+use pumpkin_core::math::position::WorldPosition;
+use pumpkin_core::math::vector3::Vector3;
+use pumpkin_macros::server_packet;
+use serde::Deserialize;
+
+use crate::{
+    bytebuf::{ByteBuffer, DeserializerError},
+    FixedBitSet, ServerPacket, VarInt,
+};
+
+#[server_packet("play:sign_update")]
+pub struct SUpdateSign {
+    pub location: WorldPosition,
+    pub is_front_text: bool,
+    pub line_1: String,
+    pub line_2: String,
+    pub line_3: String,
+    pub line_4: String,
+}
+
+impl ServerPacket for SUpdateSign {
+    fn read(bytebuf: &mut ByteBuffer) -> Result<Self, DeserializerError> {
+        let location = WorldPosition(Vector3::new(
+            bytebuf.get_i32()?,
+            bytebuf.get_i32()?,
+            bytebuf.get_i32()?,
```

Comment:
This won't work. Look at https://wiki.vg/Protocol#Position. My solution:

pumpkin-core/src/math/position impl WorldPosition add
```
pub fn from_i64(encoded_position: i64) -> Self {
        WorldPosition(Vector3 {
            x: (encoded_position >> 38) as i32,
            y: (encoded_position << 52 >> 52) as i32,
            z: (encoded_position << 26 >> 38) as i32,
        })
    }
```
And in your SUpdateSign you just do: 

`let location = WorldPosition::from_i64(bytebuf.get_i64()?);`

Reply:
Added this, thanks for the solution!